src/server/pkg/lokiutil/lokiutil.go

```diff
@@ -0,0 +1,45 @@
+package lokiutil
+
+import (
+	"sort"
+
+	"github.com/pachyderm/pachyderm/src/client/pkg/errors"
+	"github.com/pachyderm/pachyderm/src/server/pkg/errutil"
+
+	"github.com/grafana/loki/pkg/loghttp"
+)
+
+func ForEachLine(resp *loghttp.QueryResponse, f func(line string) error) error {
+	// sort and display entries
+	streams, ok := resp.Data.Result.(loghttp.Streams)
+	if !ok {
+		return errors.Errorf("respo.ResultValue must be of type loghttp.Streams to call ForEachStream on it")
+	}
+	allEntries := make([]streamEntryPair, 0)
+
+	for _, s := range streams {
+		for _, e := range s.Entries {
+			allEntries = append(allEntries, streamEntryPair{
+				entry:  e,
+				labels: s.Labels,
+			})
+		}
+	}
+
+	sort.Slice(allEntries, func(i, j int) bool { return allEntries[i].entry.Timestamp.Before(allEntries[j].entry.Timestamp) })
```

Comment:
logs aren't already sorted by timestamp?

Reply:
Yeah it's kinda lame, I think it has to do with when you have multiple Loki servers up and they don't interleave the results. I actually just copied this out of their command line client, so it seems like it's something that has to be done clientside.